### PR TITLE
[Merged by Bors] - refactor: remove some dependence on `circle` in C⋆-algebras

### PR DESCRIPTION
**Pull request by j-loreaux**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/CstarAlgebra/ContinuousFunctionalCalculus/Unitary.lean [hunk 1]
```diff
@@ -11,43 +11,46 @@ import Mathlib.Analysis.CstarAlgebra.ContinuousFunctionalCalculus.Unital
...
+lemma cfc_unitary_iff (f : R → R) (a : A) (ha : p a := by cfc_tac)
+    (hf : ContinuousOn f (spectrum R a) := by cfc_cont_tac) :
+    cfc f a ∈ unitary A ↔ ∀ x ∈ spectrum R a, star (f x) * f x = 1 := by
+  simp only [unitary, Submonoid.mem_mk, Subsemigroup.mem_mk, Set.mem_setOf_eq]
+  rw [← IsStarNormal.cfc_map (p := p) a f |>.star_comm_self |>.eq, and_self, ← cfc_one R a,
+    ← cfc_star, ← cfc_mul .., cfc_eq_cfc_iff_eqOn]
+  exact Iff.rfl
+
+end Generic
+section Complex
```
Comment (j-loreaux) [inline Mathlib/Analysis/CstarAlgebra/ContinuousFunctionalCalculus/Unitary.lean:34]:
```suggestion
end Generic

section Complex
```

**Review by YaelDillies — APPROVED** [review]
Per the discussion over at #15116, this is a sensible change.

maintainer merge

If you want to add deprecations quickly, feel free to. I personally think it's unnecessary here since CFC and friends are so recent in Mathlib and still under development by you.

**Comment by riccardobrasca** [comment]
Thanks!

bors d+

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**